### PR TITLE
app-laptop/msi-keyboard: New package

### PR DESCRIPTION
**Pull request by Fulgurance**
Control backlight of MSI laptop keyboards

Package-Manager: Portage 2.3.66, Repoman 2.3.12
Signed-off-by: Zohran Londais zohran.londais@gmail.com

### CODE REVIEW

**Comment by juippis** [comment]
Hey, I see you have multiple PRs open for new packages. None of them really comply with our guidelines. First of all, you need to sign-off all your commits, 
https://www.gentoo.org/glep/glep-0076.html#certificate-of-origin

then those commits need to be formatted to follow style, 
https://www.gentoo.org/glep/glep-0066.html#commit-messages

Also please follow the default skel type to keep unified and dont make ebuild phase functions multi-liners. So (example) `src_install() {` instead of making it double-liner. 

If you still wish to work on these ebuilds, please update them, or open new PRs! It'd be nice if you rebased against fresh ::gentoo master.

**Comment by Fulgurance** [comment]
Okay, thanks for your advices. If in my next commit i sign off, is it good or i need to edit all of my commits ?

**Comment by juippis** [comment]
You need to rebase and amend your previous commits, adding sign-off to **every** commit you make. You can automate it in the future by adding `DCO_SIGNED_OFF_BY` variable into your make.conf, check *man 5 make.conf* for correct syntax.

**Comment by Fulgurance** [comment]
How can I edit my previous commit ? I don’t see any edit button. Screenshot ?

**Comment by juippis** [comment]
You use git-tools from the command line. Don't use Github web UI.
https://wiki.gentoo.org/wiki/Gentoo_git_workflow
https://wiki.gentoo.org/wiki/Gentoo_git_workflow#Repository_settings
https://wiki.gentoo.org/wiki/Gentoo_git_workflow#Workflow_walkthrough

**Comment by Fulgurance** [comment]
It's good, i have signed all of my commits. Just i need to update now my ebuild. Ask me if you have any problem wit my edited commits

**Comment by Fulgurance** [comment]
Why i have this errors ?

**Comment by juippis** [comment]
Your copy of master branch is too old, please update the master branch and then rebase mksi-keyboard against that, then force-push an update.

**Comment by Fulgurance** [comment]
I think now it's good :)

**Comment by Fulgurance** [comment]
I have little question, after my package was accepted, can i have rights to write documentation for my package in Gentoo web site documentation ?

**Comment by juippis** [comment]
Now, please rebase and squash your commits into one, while following guidelines written in here: https://www.gentoo.org/glep/glep-0066.html#commit-messages 

> I have little question, after my package was accepted, can i have rights to write documentation for my package in Gentoo web site documentation ?

Yes, having a wiki page for it would be nice.

**Comment by Fulgurance** [comment]
I don't understand. Do i need to make just one commit for all of my commits ?

**Comment by juippis** [comment]
Yes please, all this should happen within a single commit. Also your sign-off is missing `< >` marks around your e-mail.

**Comment by Fulgurance** [comment]
Is it good ?:)

File: app-laptop/msi-keyboard/metadata.xml [hunk 1]
```diff
@@ -0,0 +1,13 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!DOCTYPE pkgmetadata SYSTEM "http://www.gentoo.org/dtd/metadata.dtd">
+<pkgmetadata>
+	<maintainer type="person">
+		<email>zohran.londais@gmail.com</email>
+	</maintainer>
+	<longdescription lang="en">
```
Comment (juippis) [inline app-laptop/msi-keyboard/metadata.xml]: You'll need to add someone with @gentoo.org email to commit for you. It's usually the proxy-maint project, 
https://wiki.gentoo.org/wiki/Project:Proxy_Maintainers/User_Guide#Proxied_maintainer_in_metadata.xml

Check examples from other proxy-maintained ebuilds.

File: app-laptop/msi-keyboard/metadata.xml [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+	Control backlight of MSI laptop keyboards without node.js
```
Comment (juippis) [inline app-laptop/msi-keyboard/metadata.xml]: This isn't really any longer than the default description and should be dropped.

File: app-laptop/msi-keyboard/msi-keyboard-1.0.ebuild [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+DESCRIPTION="Control backlight of MSI laptop keyboards"
+HOMEPAGE="https://github.com/makkarpov/msi-keyboard"
+KEYWORDS="~amd64"
+LICENSE="GPL-3"
+SLOT="0"
+RDEPEND="dev-libs/hidapi"
+PATCHES=(
+		"${FILESDIR}"/${P}-cflags.patch
+)
+SRC_URI="https://github.com/makkarpov/msi-keyboard/archive/v${PV}.tar.gz -> ${P}.tar.gz"
```
Comment (juippis) [inline app-laptop/msi-keyboard/msi-keyboard-1.0.ebuild]: Your variables are all over the place, could you follow the default skel style? 
https://github.com/gentoo/gentoo/blob/master/skel.ebuild
```
# Copyright 2019 Gentoo Authors
# Distributed under the terms of the GNU General Public License v2

EAPI=7

DESCRIPTION=""
HOMEPAGE=""
SRC_URI=""

LICENSE=""
SLOT="0"
KEYWORDS="~amd64 ~x86"
IUSE=""

DEPEND=""
RDEPEND="${DEPEND}"
BDEPEND=""
```

**Comment by juippis** [comment]
Then, you still don't follow the commit guidelines of 
https://www.gentoo.org/glep/glep-0066.html#commit-messages

https://gitweb.gentoo.org/repo/gentoo.git/log/ should also give you examples how to correctly format your git summary, and there's a lot of "spam" in your git message log :)

File: app-laptop/msi-keyboard/metadata.xml (hunk 1 above)
Comment (Fulgurance) [reply app-laptop/msi-keyboard/metadata.xml]: Can i add anybody, or i need to ask somebody if he is agree to commit for me ?

Comment (juippis) [reply app-laptop/msi-keyboard/metadata.xml]: Yes, you need your committers consent. If you dont have anyone in mind, add the proxy-maint project (see the link I gave you above)

**Comment by Fulgurance** [comment]
I think now it's good :)

**Comment by juippis** [comment]
Almost! Nearly there :)

```
 * QA Notice: Files built without respecting LDFLAGS have been detected
 *  Please include the following list of files in your report:
 * /usr/bin/msi-keyboard
```

You may have to add a patch for Makefile to respect LDFLAGS from users make.conf.
https://wiki.gentoo.org/wiki/Package_testing#make.conf

File: app-laptop/msi-keyboard/msi-keyboard-1.0.ebuild [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+HOMEPAGE="https://github.com/makkarpov/msi-keyboard"
+SRC_URI="https://github.com/makkarpov/msi-keyboard/archive/v${PV}.tar.gz -> ${P}.tar.gz"
+
+LICENSE="GPL-3"
+SLOT="0"
+KEYWORDS="~amd64"
+
+RDEPEND="dev-libs/hidapi"
+PATCHES=(
+		"${FILESDIR}"/${P}-cflags.patch
```
Comment (juippis) [inline app-laptop/msi-keyboard/msi-keyboard-1.0.ebuild]: One extra tab.

Comment (Fulgurance) [reply app-laptop/msi-keyboard/msi-keyboard-1.0.ebuild]: I think it's good, i have removed extra tab. Say me if it's bad. I use KDE with kate to edit my ebuild, i don't know what is the good tab size for gentoo ebuild

Comment (Fulgurance) [reply app-laptop/msi-keyboard/msi-keyboard-1.0.ebuild]: I have removed old patch and add new patch with cflag and ldflag

**Comment by juippis** [comment]
Did you test your ebuild?
```
 * Applying msi-keyboard-1.0-makefile.patch ...
The text leading up to this was:
--------------------------
|--- Makefile.old	2019-08-19 22:40:43.258528097 +0200
|+++ Makefile	2019-08-19 22:39:37.861531715 +0200
--------------------------
No file to patch.  Skipping patch.
2 out of 2 hunks ignored                                                      [ !! ]
 * ERROR: app-laptop/msi-keyboard-1.0::testworld failed (prepare phase):
 *   patch -p1  failed with /tmp/portage/app-laptop/msi-keyboard-1.0/files/msi-keyboard-1.0-makefile.patch
```

**Comment by Fulgurance** [comment]
Yes I have tested. Yesterday, I wanted to correct my ebuild, but my body need to sleep 😂

**Comment by Fulgurance** [comment]
It's good ! Sorry for my bad patch

**Comment by juippis** [comment]
Thanks for your patience! Welcome aboard :)

**Comment by Fulgurance** [comment]
Thanks you very much !

**Pull request closed without merging**